### ISSUE TITLE
[FEATURE] Competitions solutions

### ISSUE THREAD

**Issue by KukretiShubham** [TheAlgorithms/C-Plus-Plus#1576]
### Detailed description

As we all know CPP is quite popular among the competitive programming coders. And a lot of programming challenges are going on. There are very less places where we get solutions or approach to those questions. We will accept the solution after the competition gets over. 

### Context

This will help budding coders. 
We can include only standard platform/competitions solutions like 
1. Codeforces.
2. Google code jam.
3. Google HashCode.
4. Google Kickstart.

This will also encourage community folks to participate and contribute to the solution.

**Comment by ErrolFernandes** [comment]
happy to help with implementing this!we can also host compititons for these questions and add the best solutions into the solution directory.

**Comment by pratikgl** [comment]
> There are very less places where we get solutions or approach to those questions. 

Codeforces, Kickstarts, Facebook hacker cup, ... etc. Most of these contests post editorial after the round is over, and they are pretty decent (prepared by problem setters themselves). Streamers like SecondThread, Errichto, William lin also post detailed video solutions (not for every contest tho). Furthermore, after the completion of a round, all the received source codes are open to everyone.

**Comment by KukretiShubham** [comment]
> > There are very less places where we get solutions or approach to those questions.
> 
> Codeforces, Kickstarts, Facebook hacker cup, ... etc. Most of these contests post editorial after the round is over, and they are pretty decent (prepared by problem setters themselves). Streamers like SecondThread, Errichto, William lin also post detailed video solutions (not for every contest tho). Furthermore, after the completion of a round, all the received source codes are open to everyone.

I posted this as I couldn't find myself. Can you share links with me. 
Thanks for letting me know.

**Comment by pratikgl** [comment]
For example here is the editorial link for Kickstart Round E 2021 Question D (Increasing Sequence Card Games) : [Click here](https://codingcompetitions.withgoogle.com/kickstart/round/000000000043585c/000000000085a709#analysis)

Editorial of Codeforces Round 741 (Div. 2): [Click here](https://codeforces.com/blog/entry/94278)

Youtube channel [SecondThread](https://www.youtube.com/c/SecondThread), [William Lin](https://www.youtube.com/channel/UCKuDLsO0Wwef53qdHPjbU2Q), [Errichto](https://www.youtube.com/c/Errichto)

All the submitted solutions are also open to the public and can be accessed in the status section on codeforces. Other platforms have their own ways to access them. eg: my solution for Kickstart 2021 Round E [Click here](https://codingcompetitions.withgoogle.com/kickstart/submissions/000000000043585c/cHJhdGlrX2ds)

**Comment by KukretiShubham** [comment]
Thanks for letting me know. But, standard algorithms are also available everywhere like youtube, blogs etc & but still we are doing it to enhance it. Since we are working to enhance programming and encourage everyone to get into it. I think this works as well.

**Comment by rivalq** [comment]
@KukretiShubham  Do you mean something like portal/website, where users can submit solutions to some specific problems appearing in mentioned contests. With that other users can just query for some problem's solution in a certain language?
If yes, then if multiple users submit solutions for the same problem should we show all of them or show only some of the top solutions depending on some criterion?.

**Comment by KukretiShubham** [comment]
@rivalq my suggestion was not about creating a new portal/website. It was only about submitting the solution for the different contest. With time, folks can optimise and contribute the better solution. For checking the most optimised solution, maintainers or reviewers will have to test and check (I think this is the part we need to think about).

**Comment by rivalq** [comment]
@KukretiShubham  yeah that would also be a good idea. I would love to be part of that

**Comment by tech-nikks** [comment]
We could try to find similar questions  for the algorithms or the methods used to solve the problems for the previous contests.
like if a question is solved with a 'x' algorithm we would list the questions which were asked previously on the contests using the same algorithm


P.S:- I'd love to be part of this

**Comment by pratyaksh1610** [comment]
We can add problems related to a particular topic instead of adding competition solutions

**Comment by pratyaksh1610** [comment]
We can add problems on a particular topic or an algorithm instead of adding competition solutions, if this could make sense.

**Comment by sparsh9** [comment]
The whole dynamic programming is based on 6-7 main questions, other questions are just the derivatives of those problems. So I can include those questions.

**Comment by KukretiShubham** [comment]
Hey @sparsh9 this is a good idea. Suggest the implementation idea. Shall we create folder within this repo or a complete new repo?

**Comment by sparsh9** [comment]
@KukretiShubham This repo seems fine, a new folder will do.

**Comment by pratham1103** [comment]
I would like to work on this issue, will you please assign it to me?

**Comment by sambhavsaxena** [comment]
@sparsh9 you're on it or should i try?

**Comment by mikyoch** [comment]
Is there any min-cost-max-flow implementation in C++?
In a graph folder, there is a max-flow implementation but MCMF is not
That should be included in the repo, in my opinion

**Comment by sambhavsaxena** [comment]
@mikyoch do you mean to include the network flow algorithm?

**Comment by modiniraj09** [comment]
I want contribute to this repository?
will you please assign it to me

**Comment by mishraabhinn** [comment]
> I want contribute to this repository? will you please assign it to me

Raise a pr. It will be reviewed by maintainers.

**Comment by vinay-126** [comment]
This seems an interesting repository, i would love to work on this one, can you assign this to me..?

**Comment by Yashrajput7232** [comment]
I aslo do a lot of competative programing on platforms like  code chef , also I have give hashcode and code jam the coding contest of Google. And also know various tips and tricks. So I also want to work on this can you assign this to me?

**Comment by Panquesito7** [comment]
There's no need for an assignment: create a PR and it will get reviewed by maintainers. :)

**Comment by SORVER** [comment]
I'm looking forward to adding or enhancing competition solutions.

**Comment by realstealthninja** [comment]
feel free to start working on it

**Comment by realstealthninja** [comment]
Ive assigned you to it

**Comment by aminegh20** [comment]
Personally, having a separate repository for that is better that adding it here. This repository specializes in Standard Algorithms. I think adding competition solutions diverges a bit from the main purpose.

**Comment by realstealthninja** [comment]
> Personally, having a separate repository for that is better that adding it here. This repository specializes in Standard Algorithms. I think adding competition solutions diverges a bit from the main purpose.

i dont quite agree since python and other algorithms repository has had leet code and other such competition solutions

**Comment by sudheerxdev** [comment]
happy to help with implementing this!we can also host compititons for these questions and add the best solutions into the solution directory.
